src/wasmfx/imports.wat.pp

```diff
@@ -6,9 +6,9 @@
 
   ;; We must make sure that there is only a single memory, so that our
   ;; load/store instructions act on the C heap, not a separate memory.
-  (import "example" "memory" (memory $0 2))
+  (import WHOLE_PROGRAM_NAME "memory" (memory $0 2))
```

Comment:
I don't see the point in making this configurable. The name picked here only affects the module name we pass into `wasm-merge`. It should be sufficient to call this something fixed like "main_program" here, without the need to complicate the build process unnecessarily by substituting this placeholder.

Reply:
It is useful when including this as submodule in other repos. One may want to differentiate the type of programs.

Reply:
Do you have an example for this?

Reply:
In one case it is an example, in another it is a benchmark. It could also be the name of the application, e.g. "My_extremely_potent_webserver". The name is visible when dumping.

Reply:
I don't see why we should insist on hardcoding a particular name...

Reply:
I'm mostly just concerned with not making the compilation unnecessarily complex, given that the module in the `.wat` file is always merged with what must be the module corresponding to the main program. But this is not a hill I'm willing to die on.

Reply:
I mean like, suppose there will be more users than the two of us, then they will have a name for their application. It would be silly if they had to rename their application, because we have fixed a name for them.

Reply:
Why would they rename their application? The only place where the name that we import from in the wat file matters is the one we pass to `wasm-merge`.

Reply:
Yes, that name is the name of the whole application...

Reply:
But that string only appears in the shell command invoking `wasm-merge` and nowhere else (in particular: not the produced output file, or the input .wasm file produced by `clang`). Maybe we are talking past each other what constitutes the "name" of the application.

Reply:
I changed it `main` in the latest commit.